src/main/java/org/opentripplanner/middleware/utils/NotificationUtils.java

```diff
@@ -109,34 +128,56 @@ public static VerificationCheck checkSmsVerificationCode(String phoneNumber, Str
     }
 
     /**
-     * Send a text-only notification email for {@link OtpUser}, ensuring the correct from
+     * Send notification email for {@link OtpUser}, ensuring the correct from
      * email address is used (i.e., {@link #FROM_EMAIL}).
      */
-    public static boolean sendTextOnlyEmail(OtpUser otpUser, String subject, String text) {
-        return sendEmailViaSparkpost(FROM_EMAIL, otpUser.email, subject, text, null);
+    public static boolean sendEmail(
+        OtpUser otpUser,
+        String subject,
+        String textTemplate,
+        String htmlTemplate,
+        Map<String, Object> data
+    ) {
+        return sendEmail(FROM_EMAIL, otpUser.email, subject, textTemplate, htmlTemplate, data);
     }
 
-    /**
-     * Send notification email for {@link OtpUser}, ensuring the correct from
-     * email address is used (i.e., {@link #FROM_EMAIL}).
-     */
-    public static boolean sendEmail(OtpUser otpUser, String subject, String text, String html) {
-        return sendEmailViaSparkpost(FROM_EMAIL, otpUser.email, subject, text, html);
+    public static boolean sendEmail(
+        AdminUser adminUser,
+        String subject,
+        String textTemplate,
+        String htmlTemplate,
+        Map<String, Object> data
+    ) {
+        return sendEmail(OTP_ADMIN_DASHBOARD_EMAIL, adminUser.email, subject, textTemplate, htmlTemplate, data);
     }
 
     /**
      * Send notification email for {@link AdminUser}, ensuring the correct from
      * email address is used (i.e., {@link #OTP_ADMIN_DASHBOARD_EMAIL}).
      */
-    public static boolean sendEmail(AdminUser adminUser, String subject, String text, String html) {
-        return sendEmailViaSparkpost(OTP_ADMIN_DASHBOARD_EMAIL, adminUser.email, subject, text, html);
+    private static boolean sendEmail(
+        String fromEmail,
+        String toEmail,
+        String subject,
+        String textTemplate,
+        String htmlTemplate,
+        Map<String, Object> data
```

Comment:
This should be an Object because according to the [FreeMarker docs](https://freemarker.apache.org/docs/pgui_quickstart_createdatamodel.html) it is possible to use...

>  any kind of public class that has public getXxx/isXxx methods as prescribed by the JavaBeans specification